File: playwright-java-e2e-template/src/main/java/pages/BasePage.java
package pages;

import com.microsoft.playwright.Page;
import config.Config;

public class BasePage {
    protected final Page page;

    public BasePage(Page page) {
        this.page = page;
    }

    public void navigate(String path) {
        page.navigate(Config.getBaseUrl() + path);
    }

    public void waitForSelector(String selector) {
        page.waitForSelector(selector);
    }
}

File: swag-labs/playwright-cs/Pages/BasePage.cs
using Microsoft.Playwright;

namespace Pages;

// Classe base para páginas, contendo operações comuns de navegação e waits.
public class BasePage
{
    protected readonly IPage Page;

    public BasePage(IPage page)
    {
        Page = page;
    }

    public async Task NavigateToAsync(string path = "/")
    {
        await Page.GotoAsync(Config.BaseUrl.TrimEnd('/') + path);
    }
}

